service/matching/handler/engine.go

```diff
@@ -1425,6 +1428,82 @@ func (e *matchingEngineImpl) isShuttingDown() bool {
 	}
 }
 
+func (e *matchingEngineImpl) domainChangeCallback(nextDomains []*cache.DomainCacheEntry) {
```

Comment:
nit side note: Domain cache calls these callbacks one by one and waits for them to return. This is the one and only domain change callback in a matching service instance so sync processing here is fine. It's a bigger problem on history side that all the callbacks have to be invoked sequentially before domain cache continues its work. This can potentially delay processing of new domain updates. Something to consider revisiting at some point if we decide to shorten domain cache refresh interval